src/app/ipv/middleware.ts

```diff
@@ -340,17 +301,7 @@ const validateSessionAndPage = async (
   }
 
   if (!req.session?.ipvSessionId) {
-    logError(
-      req,
-      {
-        pageId: pageId,
-        expectedPage: req.session?.currentPage,
-      },
-      "req.ipvSessionId is null",
-    );
-
-    renderTechnicalError(req, res);
-    return false;
+    throw new UnauthorizedError("ipvSessionId is missing");
```

Comment:
There is a change here - we'll now render a 401 with a 'session ended' page, rather than a 401 with a 'technical error' page.

This feels more appropriate to me, given that the error here is ipvSessionId missing. It does happen a lot in prod, but I'm not totally sure why. It seems to happen a lot at the end of the journey - perhaps the user pressing back after being redirected to the RP?